internal/backend/remote-state/http/backend.go

```diff
@@ -107,6 +127,48 @@ type Backend struct {
 	client *httpClient
 }
 
+// configureTLS configures TLS when needed; if there are no conditions requiring TLS, no change is made.
+func (b *Backend) configureTLS(client *retryablehttp.Client, data *schema.ResourceData) error {
+	// If there are no conditions needing to configure TLS, leave the client untouched
+	skipCertVerification := data.Get("skip_cert_verification").(bool)
+	clientCACertificatePem := data.Get("client_ca_certificate_pem").(string)
+	clientCertificatePem := data.Get("client_certificate_pem").(string)
+	clientPrivateKeyPem := data.Get("client_private_key_pem").(string)
+	if !skipCertVerification && clientCACertificatePem == "" && clientCertificatePem == "" && clientPrivateKeyPem == "" {
+		return nil
+	}
+
+	// TLS configuration is needed; create an object and configure it
+	var tlsConfig tls.Config
+	client.HTTPClient.Transport.(*http.Transport).TLSClientConfig = &tlsConfig
+
+	if skipCertVerification {
+		// ignores TLS verification
+		tlsConfig.InsecureSkipVerify = true
+	}
+	if clientCACertificatePem != "" {
+		// trust servers based on a CA
+		caCertificateData, err := os.ReadFile(clientCACertificatePem)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", clientCACertificatePem, err)
+		}
+		tlsConfig.RootCAs = x509.NewCertPool()
+		if !tlsConfig.RootCAs.AppendCertsFromPEM(caCertificateData) {
+			return fmt.Errorf("failed to append certs from file %s: %w", clientCACertificatePem, err)
+		}
+	}
+	if clientCertificatePem != "" && clientPrivateKeyPem != "" {
```

Comment:
Might be helpful to throw an error if `clientCertificatePem` is set but `clientPrivateKeyPem` is not, or vice versa, as this is misconfiguration.